### PR TITLE
Added CMake build system

### PR DESCRIPTION
**Pull request by alexey-lysiuk**
Build game and tools, the latter can be disabled
Search for SDL2, paths can be confiigured
Detect MMX support in target compiler

### CODE REVIEW

**Comment by alexey-lysiuk** [comment]
I can update Travis CI configuration to use CMake.
This allows to build with several versions of GCC and Clang on Linux, and to add macOS target in the future.

Also, if you are interested I can add AppVeyor too. You need to enable it first for your repository.

File: PanzerChasm/FindSDL2.cmake [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+	# For threads, as mentioned Apple doesn't need this.
+	# In fact, there seems to be a problem if I used the Threads package
+	# and try using this line, so I'm just skipping it entirely for OS X.
+	IF(NOT APPLE)
+		SET(SDL2_LIBRARY_TEMP ${SDL2_LIBRARY_TEMP} ${CMAKE_THREAD_LIBS_INIT})
+	ENDIF(NOT APPLE)
+
+	# For MinGW library
+	IF(MINGW)
+		SET(SDL2_LIBRARY_TEMP ${MINGW32_LIBRARY} ${SDL2_LIBRARY_TEMP})
```
Comment (Panzerschrek) [inline PanzerChasm/FindSDL2.cmake:160]: This brokes my build with MinGW.

> C:/lnk/SDL2-2.0.3/lib/x86/SDL2main.lib(./Release/SDL_windows_main.obj):(.text[_main]+0x0): multiple definition of `main'
> C:/Qt5.3.0/Tools/mingw482_32/bin/../lib/gcc/i686-w64-mingw32/4.8.2/../../../../i686-w64-mingw32/lib/../lib/libmingw32.a(lib32_libmingw32_a-crt0_c.o):crt0_c.c:(.text.startup+0x0): first defined here

File: PanzerChasm/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+
+set(SAFE_CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mmmx")
+endif()
+
+CHECK_CXX_SOURCE_COMPILES("#include <mmintrin.h>
+    int main(void) { __m64 v = _m_from_int(0); }"
+    HAVE_MMX)
```
Comment (Panzerschrek) [inline PanzerChasm/CMakeLists.txt]: Replace spaces with tabs here and below.

**Review by Panzerschrek — COMMENTED** [review]
All, except linking problem on MinGW is fine.
Tomorrow i will check build also on GNU/Linux.

**Comment by alexey-lysiuk** [comment]
Fixed spaces and switched to CMake in Travis configuration.
I’ll made the single clean commit later.

File: PanzerChasm/FindSDL2.cmake (hunk 1 above)
Comment (alexey-lysiuk) [reply PanzerChasm/FindSDL2.cmake:160]: Works fine for me using MinGW 4.8.2 installed with the latest Qt Creator. SDL version is 2.0.7 though.
What are the values of SDL2_LIBRARY and SDL2MAIN_LIBRARY variables in your CMake build configuration?

Comment (Panzerschrek) [reply PanzerChasm/FindSDL2.cmake:160]: Oh, crap.
I used SDL2 for Visual Studio. With SDL2 for MinGW all works fine.

**Comment by alexey-lysiuk** [comment]
I've pushed clean version.

**Pull request merged**